lib/main/atom/views/mainPanelView.ts

```diff
@@ -205,14 +206,19 @@ export class MainPanelView extends view.View<any> {
 
     ///////////// Change JS File Status
     updateFileStatus(filePath: string) {
-        let status = getFileStatus(filePath);
-        this.fileStatus.removeClass('icon-x icon-check text-error text-success');
-        if (status.emitDiffers || status.modified) {
-            this.fileStatus.text('Js emit is outdated');
-            this.fileStatus.addClass('icon-x text-error');
+        let project = getOrCreateProject(filePath);
```

Comment:
`getOrCreateProject` is something that shouldn't be called from the front end code. We have a seperate nodejs process running the background managing projects / language server etc. More docs : https://github.com/TypeStrong/atom-typescript/blob/master/CONTRIBUTING.md#architecture